content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry/index.md

```diff
@@ -0,0 +1,247 @@
+---
+title: Collecting file-based Java logs with OpenTelemetry
+linkTitle: Collecting file-based Java logs with OpenTelemetry
+date: 
+author: >
+  [Cyrille Le Clerc](https://github.com/cyrille-leclerc) (Grafana Labs), [Gregor Zeitlinger](https://github.com/zeitlinger) (Grafana Labs)
+issue: https://github.com/open-telemetry/opentelemetry.io/issues/5606
+sig: Java, Specification
+# prettier-ignore
+cSpell:ignore: Cyrille Clerc Gregor Zeitlinger SLF4J Logback logback stdout 
+---
+Sending Java logs with OpenTelemetry typically uses the OTLP protocol. However, some scenarios require logs to be output to files or stdout due to organizational or reliability needs.
+
+A common approach to centralize logs is to use unstructured logs, parse them with regular expressions, and add contextual attributes.
+
+However, regular expression parsing is problematic. While initially easy, it becomes complex and fragile when handling all log fields, line breaks in exceptions, and unexpected log format changes. Parsing errors are inevitable with this method.
+
+## Solution
+
+The OpenTelemetry Java Instrumentation agent now offers an easy solution to convert logs from frameworks like SLF4J/Logback or Log4j2 into OTel-compliant JSON logs on stdout with all resource and log attributes.
+
+This is a true turnkey solution:
+
+* No code or dependency changes, just a few configuration adjustments typical for production deployment.
+* No complex field mapping in the log collector, just use the [OTLP/JSON connector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/otlpjsonconnector) to ingest the payload.
+* Automatic correlation between logs, traces, and metrics.
+
+This blog post shows how to set up this solution step by step.
+
+* In the first part, we'll show how to configure the Java application to output logs in the OTLP/JSON format. 
+* In the second part, we'll show how to configure the OpenTelemetry Collector to ingest the logs.
+* Finally, we'll show a Kubernetes-specific setup to handle container logs.
+
+## Reference architecture
+
+The deployment architecture looks like the following:
+
+![OTLP/JSON Architecture](otlpjson-architecture.png)
+
+## Configure Java Application to Output OTLP/JSON Logs
+
+Steps
+
+* No code changes needed. Continue using your preferred logging library, including templated logs, mapped diagnostic context, and structured logging.
```

Comment:
Remove the "Steps" and remove the bullet points

Reply:
```suggestion
No code changes needed. Continue using your preferred logging library, including templated logs, mapped diagnostic context, and structured logging.
```